src/zenml/integrations/pandas/materializers/pandas_materializer.py

```diff
@@ -0,0 +1,192 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Materializer for Pandas."""
+
+import os
+from typing import Any, ClassVar, Dict, Optional, Tuple, Type, Union
+
+import pandas as pd
+
+from zenml.artifact_stores.base_artifact_store import BaseArtifactStore
+from zenml.enums import ArtifactType, VisualizationType
+from zenml.logger import get_logger
+from zenml.materializers.base_materializer import BaseMaterializer
+from zenml.metadata.metadata_types import DType, MetadataType
+
+logger = get_logger(__name__)
+
+PARQUET_FILENAME = "df.parquet.gzip"
+COMPRESSION_TYPE = "gzip"
+
+CSV_FILENAME = "df.csv"
+
+
+class PandasMaterializer(BaseMaterializer):
+    """Materializer to read data to and from pandas."""
+
+    ASSOCIATED_TYPES: ClassVar[Tuple[Type[Any], ...]] = (
+        pd.DataFrame,
+        pd.Series,
+    )
+    ASSOCIATED_ARTIFACT_TYPE: ClassVar[ArtifactType] = ArtifactType.DATA
+
+    def __init__(
+        self, uri: str, artifact_store: Optional[BaseArtifactStore] = None
+    ):
+        """Define `self.data_path`.
+
+        Args:
+            uri: The URI where the artifact data is stored.
+            artifact_store: The artifact store where the artifact data is stored.
+        """
+        super().__init__(uri, artifact_store)
+        try:
+            import pyarrow  # type: ignore # noqa
+
+            self.pyarrow_exists = True
+        except ImportError:
+            self.pyarrow_exists = False
+            logger.warning(
+                "By default, the `PandasMaterializer` stores data as a "
+                "`.csv` file. If you want to store data more efficiently, "
+                "you can install `pyarrow` by running "
+                "'`pip install pyarrow`'. This will allow `PandasMaterializer` "
+                "to automatically store the data as a `.parquet` file instead."
+            )
```

Comment:
Why not offer other options like `fastparquet`?

Reply:
I have not changed the implementation itself. Just moved it to an integration. If we want to offer other options, I would handle this in a separate PR.